lib/widgets/emoji_reaction.dart

```diff
@@ -614,3 +627,411 @@ class EmojiPickerListEntry extends StatelessWidget {
+      controlsNodes: {ViewReactionsUserList.semanticsIdentifier},
+
+      selected: selected,
+      label: zulipLocalizations.seeWhoReactedSheetEmojiNameWithVoteCount(emojiName, count),
+      onTap: onRequestSelect,
+      child: ExcludeSemantics(
+        child: result));
+  }
+}
+
+
+@visibleForTesting
+class ViewReactionsUserList extends StatelessWidget {
+  const ViewReactionsUserList(this.pageContext, {
+    super.key,
+    required this.messageId,
+    required this.reactionType,
+    required this.emojiCode,
+    required this.emojiName,
+  });
+
+  final BuildContext pageContext;
+  final int messageId;
+  final ReactionType reactionType;
+  final String emojiCode;
+  final String emojiName;
+
+  static const semanticsIdentifier = 'view-reactions-user-list';
+
+  @override
+  Widget build(BuildContext context) {
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final designVariables = DesignVariables.of(context);
+
+    final message = store.messages[messageId];
+
+    final userIds = message?.reactions?.aggregated.firstWhereOrNull(
+      (x) => x.reactionType == reactionType && x.emojiCode == emojiCode
+    )?.userIds.toList();
+
+    // (No filtering of muted or deactivated users.)
+
+    if (userIds == null) {
+      // This reaction lost all its votes, or the message was deleted.
+      return SizedBox.shrink();
+    }
+
+    // TODO sort userIds?
+
+    Widget result = InsetShadowBox(
+      top: 8, bottom: 8,
+      color: designVariables.bgContextMenu,
+      child: SizedBox(
+        height: 400, // TODO(design) tune
+        child: ListView.builder(
+          padding: EdgeInsets.symmetric(vertical: 8),
+          itemCount: userIds.length,
+          itemBuilder: (context, index) =>
+            ViewReactionsUserItem(context, userId: userIds[index]))));
+
+    return Semantics(
+      identifier: semanticsIdentifier, // See note on `controlsNodes` on the tab.
+      label: zulipLocalizations.seeWhoReactedSheetUserListLabel(emojiName, userIds.length),
+      role: SemanticsRole.tabPanel,
+      container: true,
+      child: result);
+  }
+}
+
+@visibleForTesting
+class ViewReactionsUserItem extends StatelessWidget {
+  const ViewReactionsUserItem(this.pageContext, {
+    super.key,
+    required this.userId,
+  });
+
+  final BuildContext pageContext;
+  final int userId;
+
+  void _onPressed() {
+    // Dismiss the action sheet.
+    Navigator.pop(pageContext);
+
+    Navigator.push(pageContext,
+      ProfilePage.buildRoute(context: pageContext, userId: userId));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    final designVariables = DesignVariables.of(context);
+
+    return InkWell(
+      onTap: _onPressed,
+      splashFactory: NoSplash.splashFactory,
+      overlayColor: WidgetStateColor.resolveWith((states) =>
+        states.any((e) => e == WidgetState.pressed)
```

Comment:
```suggestion
        states.contains(WidgetState.pressed)
```

Looks like it's a Set, so that should be equivalent (and potentially faster)